### ISSUE TITLE
Important Warnings are silenced by default

### ISSUE THREAD

**Issue by ajanvrin** [pimalaya/himalaya#522]
First of all, thank you very much for developing Himalaya, I've been looking for such a tool for a long time, and it fits perfectly what I was after!

Now, I've been using it so sift through tens of thousands of mails today, and there are some rough angles, I have seen a few errors, warnings and panics that may warrant a few bug reports if I can isolate and pinpoint them.

One thing that might bite a few users though, is that **Warnings are not displayed by default**.
I understand that INFO level logging is not enabled by default, as the tool aims for an uncluttered output, but Warnings convey important messages that the users may want to see even if they did not enable --debug or --trace.

Just as I was scratching my head trying why himalaya was only showing a single mail in my inbox whereas the other mail client showed a few thousands, I turned on the --debug flag, and realized that there were 2828 instances of WARN fetch_envelopes_by_sequence{client=1}: imap_client::tasks: skipping invalid fetch fetch="[...]" in the logs.
(I'll probably make a separate issue for that, if I find a probable cause, or after failing to find one)

To me, it is particularly important to display these warnings, because **a folder may seem empty, enticing a user to delete it, which may lead to the involutary deletion of thousands of mails**.

Fortunately, it is unlikely that this resulted in involuntary mail deletion in my case, but it might have in other circumstances.

## Expected Behavior
I'd expect WARN level messages to be displayed by default.

## Current Behavior
WARN level log messages are silent unless --debug or --trace is turned on.
Messages such as "WARN fetch_envelopes_by_sequence{client=1}: imap_client::tasks: skipping invalid fetch" are silenced, which may lead to folders seeming empty when they are not, potentially leading to involuntary mail deletion.

## Possible Solution
- Print WARN level messages by default
- Maybe add a configuration option in the TOML config to silence them by default
- Or a --quiet flag that would reproduce the current behavior

## Steps to Reproduce
Omit --debug or --trace and use himalaya in a way that will produce a Warning.
I have yet to find a reliable way to reproduce the specific WARN fetch_envelopes_by_sequence{client=1}: imap_client::tasks: skipping invalid fetch I'm encountering, I'll work on that.

Best regards

**Comment by soywod** [comment]
> First of all, thank you very much for developing Himalaya, I've been looking for such a tool for a long time, and it fits perfectly what I was after!

Happy to hear that, thanks a lot!

> One thing that might bite a few users though, is that **Warnings are not displayed by default**.

It's a very good point. It used to be the case, but warnings were destroying the view (inserting in the middle of the table for example) so I ended up not showing them by default.

> I turned on the --debug flag, and realized that there were 2828 instances of WARN fetch_envelopes_by_sequence{client=1}: imap_client::tasks: skipping invalid fetch fetch="[...]" in the logs.

Here some context. Himalaya switched to a fresh new IMAP lib that is quite accurate regarding RFCs, but also flexible with real life IMAP server implementations. You are not supposed to get this warning (which is closer to an error in fact): it means that sth is wrong is the server implementation. It definitely needs to be identified and fixed (either server side or lib side, by adding a new "quirk"). The more quirks the IMAP lib will develop, the better it will operate with real world IMAP servers. And less warnings we should deal with. See the [actual quirks developed by the lib](https://github.com/duesee/imap-codec/blob/e0f7c9b838030a202b8d94416f1578f68158e292/imap-codec/Cargo.toml#L21-L43).

That said, showing warnings back by default is quite appealing, I just don't know how to proceed.
The cleanest way I could possibly imagine is to collect warnings, then to display them at the end, after the content.
All other options can be discarded, because a simple alias could work (sth like `alias himalaya='RUST_LOG=warn himalaya'`).

**Comment by ajanvrin** [comment]
> It's a very good point. It used to be the case, but warnings were destroying the view (inserting in the middle of the table for example) so I ended up not showing them by default.

I think this is fine, if the user has a means to silence the warnings.
It turns out, the user can already do that in at least two ways:
- prepend `RUST_LOG=error` to the command
- append `2>/dev/null` to the command (or `2>$null` on Windows powershell)
(I was pleased to see that logs are sent to stderr and not stdout by the way, that's quite sensible)
It also works by design when a user is parsing the json output, as unless `2>&1` is added to the command, the next process in the pipe will only receive the content of stdout, which is what we want.

> Here some context. Himalaya switched to a fresh new IMAP lib that is quite accurate regarding RFCs, but also flexible with real life IMAP server implementations. You are not supposed to get this warning (which is closer to an error in fact)

Ah I see, indeed.

> The cleanest way I could possibly imagine is to collect warnings, then to display them at the end, after the content.

Well, I kinda disagree there. Displaying warnings at the end removes them from the context, and introduces unnecessary complexity: what if an error/panic happens later, would warnings be lost?
As said above, I think outputting warnings right when they appear is sensible, provided they go to stderr and not stdout.

**Comment by soywod** [comment]
> I think this is fine, if the user has a means to silence the warnings.
> It turns out, the user can already do that in at least two ways:
> - prepend `RUST_LOG=error` to the command
> - append `2>/dev/null` to the command (or `2>$null` on Windows
> powershell)

True, it should be enough to gather logs. It was not the case before,
because logs were not properly sent to stderr.

Let's keep this issue for enabling warnings by default, thank you!

**Comment by soywod** [comment]
Warn level put back by default on `pimalaya/tui/master`, it should be available on the next Himalaya CLI patch release!

**Comment by soywod** [comment]
New release [`v1.1.0`](https://github.com/pimalaya/himalaya/releases/tag/v1.1.0) out!

**Comment by ChiotOnAGE** [comment]
I think we need a `--quiet` then. We already have `--debug` and `--trace`, it make sense to have `--quiet` as well. While it is possible to suppress the warnings with redirection or a change in the environment variable, having `--quiet` seems much more consistent.

**Comment by soywod** [comment]
> I think we need a `--quiet` then.

I totally agree with your proposition, I opened a dedicated issue for that purpose: https://github.com/pimalaya/himalaya/issues/545
